File: swift/Dry Electrics/common/ChargeInfo.swift
import Foundation

class ChargeInfo
{
    var user: String
    var percent: Int
    var color: Color
    var icon: String
    var charging: Int
    
    init()
    {
        user = "N/A"
        percent = 0
        color = Color.clear
        icon = "?"
        charging = -1
    }
    
    // TODO: use dict
    init(user: Array<Any>)
    {
        color = hexStringToColor(hex: user[0] as! String)
        self.user = user[1] as! String;
        percent = user[2] as! Int;
        icon = user[3] as! String;
        charging = -1
    }
}

File: android/app/src/main/java/org/dry/electrics/ChargeInfo.kt
package com.dry.electrics

import android.graphics.Color
import org.json.JSONArray

class ChargeInfo(user: JSONArray)
{
    var user: String
    var percent: Int
    var color: Int
    var icon: String
    var charging: Int

    init
    {
        this.user = "N/A"
        percent = 0
        color = Color.TRANSPARENT
        icon = "?"
        charging = -1

        if (user.length() >= 4) {
            color = hexStringToColor(user.getString(0))
            this.user = user.getString(1)
            percent = user.getInt(2)
            icon = user.getString(3)
            charging = -1
        }
    }
}